tests/test_telemetry_queue_worker.py

```
@@ -0,0 +1,79 @@
+    caplog.set_level("DEBUG")
+
+    collector = telemetry.TelemetryCollector()
+    # Force-enable telemetry regardless of env settings from conftest
+    collector.config.enabled = True
+
+    # Replace queue with tiny one to trigger backpressure quickly
+    small_q = q.Queue(maxsize=2)
+    collector._queue = small_q
+
+    # Make sends slow to build backlog and exercise worker
+    def slow_send(self, rec):
```

Ruff (0.12.2)

Unused function argument: `self`

(ARG001)